src/humanloop/otel/exporter.py

```diff
@@ -0,0 +1,148 @@
+from queue import Queue
+from threading import Thread
+import typing
+from opentelemetry import trace
+from opentelemetry.sdk.trace import Span
+from opentelemetry.sdk.trace.export import SpanExporter, SpanExportResult
+
+from humanloop.otel.constants import HL_FILE_OT_KEY, HL_LOG_OT_KEY, HL_TRACE_METADATA_KEY, OT_EMPTY_ATTRIBUTE
+from humanloop.otel.helpers import read_from_opentelemetry_span
+
+if typing.TYPE_CHECKING:
+    from humanloop.base_client import BaseHumanloop
+
+
+class HumanloopSpanExporter(SpanExporter):
+    """SpanExporter that uploads OpenTelemetry spans to Humanloop Humanloop spans."""
+
+    WORK_THREADS = 8
+
+    def __init__(self, client: "BaseHumanloop") -> None:
+        super().__init__()
+        self._client = client
+        self._uploaded_log_ids = {}
+        self._upload_queue = Queue()
+        self._threads = [Thread(target=self._do_work, daemon=True) for _ in range(self.WORK_THREADS)]
+        self._shutdown = False
+        for thread in self._threads:
+            thread.start()
+
+    def export(self, spans: trace.Sequence[Span]) -> SpanExportResult:
+        for span in spans:
+            self._upload_queue.put(span)
+
+    def shutdown(self) -> None:
+        self._shutdown = True
+        for thread in self._threads:
+            thread.join()
+
+    def force_flush(self, timeout_millis: int = 3000) -> bool:
+        self._shutdown = True
+        for thread in self._threads:
+            thread.join(timeout=timeout_millis)
+        self._upload_queue.join()
+
+        return True
+
+    def _do_work(self):
+        # Do work while the Exporter was not instructed to
```

Comment:
use proper docstring

Reply:
ack

Reply:
acknowledged, but don't see change to docstring

Reply:
<img width="1049" alt="image" src="https://github.com/user-attachments/assets/bee1157b-455c-4c2a-b99c-e2ba037db714">
?